pkg/watches/watches.go

```diff
@@ -55,42 +74,63 @@ func Load(path string) ([]Watch, error) {
 		return nil, err
 	}
 
-	watchesMap := make(map[schema.GroupVersionKind]Watch)
+	watchesMap := make(map[schema.GroupVersionKind]struct{})
 	for i, w := range watches {
-		if err := verifyGVK(w.GroupVersionKind); err != nil {
-			return nil, fmt.Errorf("invalid GVK: %s: %w", w.GroupVersionKind, err)
+		gvk := w.GroupVersionKind
+
+		if err := verifyGVK(gvk); err != nil {
+			return nil, fmt.Errorf("invalid GVK: %s: %w", gvk, err)
+		}
+
+		if _, err := chartutil.IsChartDir(w.ChartPath); err != nil {
+			return nil, fmt.Errorf("invalid chart directory %s: %w", w.ChartPath, err)
 		}
 
 		cl, err := loader.Load(w.ChartPath)
 		if err != nil {
 			return nil, fmt.Errorf("invalid chart %s: %w", w.ChartPath, err)
 		}
 		w.Chart = cl
-		w.OverrideValues = expandOverrideEnvs(w.OverrideValues)
+
+		if _, ok := watchesMap[gvk]; ok {
+			return nil, fmt.Errorf("duplicate GVK: %s", gvk)
+		}
+
+		watchesMap[gvk] = struct{}{}
```

Comment:
Nit: keep the map assignment grouped together with the duplicate check. Having this line grouped with the `if w.WatchDependentResources == nil` check is a little weird IMO. 
```suggestion
		watchesMap[gvk] = struct{}{}

```